compiler/rustc_ast_lowering/src/pat.rs

```diff
@@ -10,7 +10,11 @@ use rustc_span::symbol::Ident;
 use rustc_span::{source_map::Spanned, Span};
 
 impl<'a, 'hir> LoweringContext<'a, 'hir> {
-    crate fn lower_pat(&mut self, mut pattern: &Pat) -> &'hir hir::Pat<'hir> {
+    crate fn lower_pat(&mut self, pattern: &Pat) -> &'hir hir::Pat<'hir> {
+        self.arena.alloc(self.lower_pat_mut(pattern))
+    }
+
+    crate fn lower_pat_mut(&mut self, mut pattern: &Pat) -> hir::Pat<'hir> {
```

Comment:
The `_mut` confused me a bit. Did you mean "owned and not arena allocated"? I think we usually go for something like `_inner`, but other suggestions are also welcome.

Reply:
I agree it's not very intuitive. I think it's consistent with existing code at least. I think of it as "use this if you want to modify the output or put it in a (mutable) container".

Reply:
oh, there's precedent for these postfixes? still confusing, but yea, no need to change that in this pr